toxcore/group.c

```diff
@@ -946,31 +955,28 @@ int find_peer_index_in_list(const Group_c *g, aint peer_index)
 
 /* Set the nick for a peer.
  *
- * return 0 on success.
- * return -1 if error.
+ * return true on success.
+ * return false if error.
  */
-static aint setnick(Group_c *g, aint peer_index, const uint8_t *nick, aint nick_len)
+static bool setnick(Group_c *g, int64_t peer_index, const uint8_t *nick, uint8_t nick_len)
 {
     if (nick_len > MAX_NAME_LENGTH || nick_len == 0) {
-        return -1;
+        return false;
     }
 
-    Group_Peer *peer = g->peers + peer_index;
+    Group_Peer *peer = &g->peers[peer_index];
 
     /* same name as already stored? */
-    if (peer->nick_len == nick_len) {
-        if (nick_len == 0 || !memcmp(peer->nick, nick, nick_len)) {
-            return 0;
-        }
+    if (peer->nick_len == nick_len && memcmp(peer->nick, nick, nick_len) == 0) {
+        return true;
     }
 
     if (nick_len) {
-
         if (nick_len > peer->nick_len) {
             uint8_t *new_nick = realloc(peer->nick, nick_len + 1);
 
             if (!new_nick) {
-                return 0;
+                return false;
```

Comment:
I made a logic change here. Previously the function returned success if it couldn't allocate space for a new nickname.